tests/foreman/cli/test_computeresource.py

```diff
@@ -299,47 +334,42 @@ def test_update_positive(self, options):
         @Assert: Compute Resource successfully updated
 
         """
-        comp_res = make_compute_resource()
-        options.update({
-            'name': comp_res['name'],
-        })
-
-        # update Compute Resource
-        result = ComputeResource.update(options)
-        self.assertEqual(result.return_code, 0)
-        self.assertEqual(len(result.stderr), 0)
-
-        # check updated values
-        result = ComputeResource.info({'id': comp_res['id']})
-        self.assertEqual(result.return_code, 0)
-        self.assertEqual(len(result.stderr), 0)
-        self.assertEqual(
-            result.stdout['description'],
-            options.get('description', comp_res['description'])
-        )
-        self.assertEqual(
-            result.stdout['name'],
-            options.get('new-name', comp_res['name'])
-        )
-        self.assertEqual(
-            result.stdout['url'],
-            options.get('url', comp_res['url'])
-        )
-        self.assertEqual(
-            result.stdout['provider'].lower(), comp_res['provider'].lower())
+        for options in valid_update_data():
+            with self.subTest(options):
+                comp_res = make_compute_resource()
```

Comment:
The function valid_update_data() supplies variety of test data like name, new-name, url, description.  By reusing compute resource, this test will look more complicated (I will have to store the old values in separate variables after every iteration). I like to leave it this way.